lib/dry/struct/class_interface.rb

```diff
@@ -67,7 +67,7 @@ def constructor_type(type = nil)
 
       def schema
         super_schema = superclass.respond_to?(:schema) ? superclass.schema : {}
-        super_schema.merge(@schema || {})
+        super_schema.merge((defined? @schema) ? @schema : {})
```

Comment:
I guess `super_schema.merge(defined?(@schema) ? @schema : {})` will be cleaner :)